lando/common/commands.py

```diff
@@ -0,0 +1,300 @@
+        self.paths = paths
+
+    def command_file_dict(self, methods_document_content):
+        additional_log_files = []
+        if self.names.usage_report_path:
+            additional_log_files.append(self.names.usage_report_path)
+        return {
+            "bespin_job_id": self.job.id,
+            "destination_dir": self.paths.OUTPUT_RESULTS_DIR,
+            "downloaded_workflow_path": self.names.workflow_download_dest,
+            "workflow_to_read": self.names.workflow_to_read,
+            "workflow_type": self.job.workflow.workflow_type,
+            "job_order_path": self.names.job_order_path,
+            "bespin_workflow_stdout_path": self.names.run_workflow_stdout_path,
+            "bespin_workflow_stderr_path": self.names.run_workflow_stderr_path,
+            "methods_template": methods_document_content,
+            "additional_log_files": additional_log_files
+        }
+
+    def run(self, base_command, methods_document_content):
+        command_filename = self.names.organize_output_command_filename
+        self.write_json_file(command_filename, self.command_file_dict(methods_document_content))
+        command = base_command.copy()
+        command.append(self.names.organize_output_command_filename)
+        self.run_command(command)
+
+
+class SaveOutputCommand(BaseCommand):
+    def __init__(self, names, paths, activity_name, activity_description):
+        self.names = names
+        self.paths = paths
+        self.activity_name = activity_name
+        self.activity_description = activity_description
+
+    def command_file_dict(self, share_dds_ids, started_on, ended_on):
+        return {
+            "destination": self.names.output_project_name,
+            "readme_file_path": self.paths.REMOTE_README_FILE_PATH,
+            "paths": [self.paths.OUTPUT_RESULTS_DIR],
+            "share": {
+                "dds_user_ids": share_dds_ids
+            },
+            "activity": {
+                "name": self.activity_name,
+                "description": self.activity_description,
+                "started_on": started_on,
+                "ended_on": ended_on,
+                "input_file_versions_json_path": self.names.workflow_input_files_metadata_path,
+                "workflow_output_json_path": self.names.run_workflow_stdout_path
+            }
+        }
+
+    def run(self, base_command, dds_credentials, share_dds_ids, started_on, ended_on):
+        command_filename = self.names.save_output_command_filename
+        self.write_json_file(command_filename, self.command_file_dict(share_dds_ids, started_on, ended_on))
+
+        dds_config_filename = self.names.dds_config_filename
+        self.write_dds_config_file(dds_config_filename, dds_credentials)
+
+        command = base_command.copy()
+        command.append(command_filename)
+        command.append(self.names.output_project_details_filename)
+        command.append("--outfile-format")
+        command.append("json")
```

Comment:
I found this confusing - I recall that the  k8s implementation works differently, but this method being in `lando.common` suggests otherwise.

Upon further digging, I see that lando.k8s doesn't call `SaveOutputCommand.run()`. Perhaps this method should be in a subclass in `lando.worker` since it's only used there?

Reply:
`lando.k8s` only uses the `command_file_dict` method for all commands:(StageDataCommand, RunWorkflowCommand, OrganizeOutputCommand, SaveOutputCommand).
`lando.worker` only uses the `run` method for all commands. It seemed more natural to write the two functions together. I could move the `run` methods into the `lando.worker` directory with subclasses if that is easier to follow.